Part1/code/LimitedColorPerception.py

```diff
@@ -1,58 +1,66 @@
-import torch
 from torch.utils.data import Dataset
 from torchvision import transforms
 from RGBAndContrastTransform import RGBAndContrastTransform
-
 from PIL import Image
 
+
 class LimitedColorPerceptionDataset(Dataset):
-    def __init__(self, dataset, month_age, max_age=12.0):
+    def __init__(self, dataset, month_age, max_age=6) -> None:
         """
-        Initializes the dataset with limited color perception based on infant color development.
-
         Args:
-            dataset (Dataset): The original dataset to wrap.
-            month_age (float): The age in months of the subject.
-            max_age (float, optional): The age in months when full color perception is reached. Defaults to 12.0.
+            month_age (int): The age in months of the person.
+            max_age (int): The age in months when full color perception is reached.
         """
         self.dataset = dataset
         self.month_age = month_age
         self.max_age = max_age
+        # Max age value is choosen as a result of trial and error
 
-        self.infant_development_stages = {
-            0: {'red': 0.8, 'green': 0.6, 'blue': 0.5, 'contrast': 0.6},
-            1: {'red': 0.84, 'green': 0.65, 'blue': 0.55, 'contrast': 0.65},
-            2: {'red': 0.88, 'green': 0.75, 'blue': 0.6, 'contrast': 0.7},
-            3: {'red': 0.92, 'green': 0.8, 'blue': 0.65, 'contrast': 0.75},
-            4: {'red': 0.95, 'green': 0.85, 'blue': 0.7, 'contrast': 0.8},
-            5: {'red': 0.98, 'green': 0.9, 'blue': 0.75, 'contrast': 0.85},
-            6: {'red': 1.0, 'green': 0.95, 'blue': 0.8, 'contrast': 0.9},
-            7: {'red': 1.0, 'green': 0.97, 'blue': 0.85, 'contrast': 0.95},
-            8: {'red': 1.0, 'green': 1.0, 'blue': 0.9, 'contrast': 1.0},
-            9: {'red': 1.0, 'green': 1.0, 'blue': 1.0, 'contrast': 1.0},
-            10: {'red': 1.0, 'green': 1.0, 'blue': 1.0, 'contrast': 1.0},
-            11: {'red': 1.0, 'green': 1.0, 'blue': 1.0, 'contrast': 1.0},
-            12: {'red': 1.0, 'green': 1.0, 'blue': 1.0, 'contrast': 1.0},
+        # Inital color properties (red, green, blue, and contrast)
```

Comment:
if the results are similar, I think this is a more cool way to implement. if not so, we can revert the changes since the cost is high.

Reply:
The cost is irrelevant, since manually creating a fixed-size dictionary is practically the same with doing it via for loop. Especially for a dictionary of a size between 6 and 12. I suggest keeping the new snippet, since the final version of this part should be provided by Rasul (considering that he is the one who provided images to the report with his own channel and contrast configuration). After merging this to the main branch, this part will be overridden by Rasul's branch.